### PR TITLE
Add an entpoint which accepts a hydra session ID, then requests the u…

### PR DESCRIPTION
**Pull request by knoxsp**
…ser details from another service, using this session id. THe idea is that another service to which a user has logged in can be referred, thus allowing them to avoid logging into the API

### CODE REVIEW

**Review by pmslavin — COMMENTED** [review]
Couple of minor comments and things I don't understand.

File: hydra_server/server/service.py [hunk 1]
```diff
@@ -163,4 +165,45 @@ def login(ctx, username, password):
...
+        if resp.ok:
+            data = json.loads(resp.content)
+            ctx.transport.req_env['beaker.session']['user_id'] = data['user_id']
+            ctx.transport.req_env['beaker.session']['username'] = data['username']
+            ctx.transport.req_env['beaker.session']['id'] = session_id
+            ctx.transport.req_env['beaker.session'].save()
+
+        login_response = LoginResponse()
+        login_response.user_id = data['user_id']
+        login_response.username = data['username']
```
[the remark this answers is not in the document]
Comment (knoxsp) [reply hydra_server/server/service.py]: Addressed

File: hydra_server/server/service.py [hunk 2]
```diff
@@ -163,4 +165,45 @@ def login(ctx, username, password):
...
+    @rpc(Unicode, _returns=LoginResponse)
+    def get_remote_session(ctx, session_id=None):
+        """
+            This function takes a session id which was not generated by this server,
+            but by another hydra server. It makes a request to the remote server using
+            the endpoint 'get_hydra_session', which returns the user ID, username and session ID.
+            From here we can set the session ID as though it was generated here.
+
+        """
+        remote_hydra_server = config.get('SSO', 'url', 'localhost').strip('/')
```
Comment (pmslavin) [inline hydra_server/server/service.py:181]: I had to add the transport to this line to get this to work.

```suggestion
        remote_hydra_server = config.get('SSO', 'url', 'http://localhost').strip('/')
```

The first attempt below failed (returning `None`) and use just `localhost` as the `remote_hydra_server`.  This gave...
```
1310735 2022-05-18 17:19:15,239 - INFO - Received request: <function AuthenticationService.get_remote_session at 0x7f112512faf0>
1310735 2022-05-18 17:19:15,241 - CRITICAL - No connection adapters were found for 'localhost:8081/get_hydra_session/97d7fb5201c24fc1b5b4a254d4b12999'
```

The second correct response is after adding `http://` as above...



```python
>>> client=RemoteJSONConnection("http://localhost:8080/json")
1310600 2022-05-18 17:10:27,603 - INFO - Using user-defined URL: http://localhost:8080/json
>>> client.login("pmslavin@gmail.com", "rfc1034")
1310600 2022-05-18 17:10:55,125 - INFO - Calling: login
1310600 2022-05-18 17:10:55,125 - INFO - Args {'login': {'username': 'pmslavin@gmail.com', 'password': 'rfc1034'}}
1310600 2022-05-18 17:10:55,691 - INFO - 97d7fb5201c24fc1b5b4a254d4b12999
1310600 2022-05-18 17:10:55,692 - INFO - done (0.5666725635528564)
1310600 2022-05-18 17:10:55,692 - INFO - Login response OK for user: 3
(3, '97d7fb5201c24fc1b5b4a254d4b12999')
>>> client.get_remote_session(session_id="97d7fb5201c24fc1b5b4a254d4b12999")
1310600 2022-05-18 17:12:14,527 - INFO - Calling: get_remote_session
1310600 2022-05-18 17:12:14,528 - INFO - Args {'get_remote_session': {'session_id': '97d7fb5201c24fc1b5b4a254d4b12999'}}
1310600 2022-05-18 17:12:14,702 - INFO - done (0.17462897300720215)
{'user_id': None}
>>> client.get_remote_session(session_id="97d7fb5201c24fc1b5b4a254d4b12999")
1310600 2022-05-18 17:15:55,951 - INFO - Calling: get_remote_session
1310600 2022-05-18 17:15:55,952 - INFO - Args {'get_remote_session': {'session_id': '97d7fb5201c24fc1b5b4a254d4b12999'}}
1310600 2022-05-18 17:15:56,138 - INFO - done (0.1869831085205078)
{'user_id': 3}
```

**Review by pmslavin — COMMENTED** [review]
Had to add a protocol to the `remote_hydra_server` var to get this to work, as per comment.
Worked successfully after that.

**Pull request merged**